addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -940,20 +949,43 @@ options.registry.OptionsTab = options.Class.extend({
     //--------------------------------------------------------------------------
     // Private
     //--------------------------------------------------------------------------
+
     /**
      * @private
      * @param {String} id
-     * @param {Object} settings: gray settings that will be adjusted to
-     * @returns {String} the new color of gray
+<     * @returns {String} the new color of gray
      */
-    _buildGray(id, settings) {
-        this['gray-hue'] = settings['gray-hue'] || this['gray-hue'];
-        this['gray-extra-saturation'] = settings['gray-extra-saturation'] || this['gray-extra-saturation'];
+    _buildGray(id) {
         const gray = weUtils.getCSSVariableValue(`base-${id}`);
         const hsl = weUtils.colorHEXtoHSL(gray);
-        hsl.h = this['gray-hue'];
-        hsl.s = Math.min(Math.max(hsl.s + this['gray-extra-saturation'], 0), 100);
-        return `hsl(${hsl.h},${hsl.s}%,${hsl.l}%)`;
+        hsl.h = this.grays['gray-hue'];
+        hsl.s = Math.min(Math.max(hsl.s + this.grays['gray-extra-saturation'], 0), 100);
+        return weUtils.colorHSLToHEX(hsl.h, hsl.s, hsl.l);
+    },
+    /**
+     * @private
+     * update UI gray palette preview
+     */
+    _updateGrayPaletteUI() {
+        this.$el.find("span#grays").each((_, e) => {
+            e.style.setProperty("background-color", this.grays[e.getAttribute('variable')], "important");
+        });
+    },
+    /**
+     * @private
+     */
+    async _customizeWebsiteGrayParameters(values, params) {
+        const url = '/website/static/src/scss/options/user_gray_palette_parameters.scss';
+        return this._makeSCSSCusto(url, values);
+    },
+    /**
+     * @override
+     */
+    async _customizeWebsiteByType(widgetValue, params, type) {
+        if (type === "grays") {
+            return this._customizeWebsiteGrayParameters(widgetValue, params);
+        }
+        return this._super(...arguments);
```

Comment:
It's ok to have split `_customizeWebsite` to do that but feels like a bit overkill. Add the possibility to customize grays in the base Class directly, it would be easier. No need to split and everything..

Reply:
IMO, if I add gray persistence logic (`_customizeWebsite`) in the base Class, I will also need to add `_computeWidgetState` logic in the base Class as well.
so should i put all the gray logic in the base Class or just persistence logic?

https://github.com/odoo/odoo/blob/0816450bb6583f4b81b96669b7e972b9990a8826/addons/website/static/src/js/editor/snippets.options.js#L1022

Reply:
Mmh, ok, let's keep the split you made :+1: